### PR TITLE
Change build path for linux binaries to match GOARCH.

### PR DESCRIPTION
**Pull request by jefchien**
**Description:** Simplifies the docker build process by changing the linux binary build paths to match GOARCH/TARGETARCH.

**Link to tracking Issue:** https://github.com/aws-observability/aws-otel-collector/issues/880

**Testing:** Ran `create_rpm.sh` and `create_deb.sh` (partially). Ran `make docker-build` and `make docker-build-arm`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,22 +78,22 @@ all-srcs:
 .PHONY: build
 build: install-tools lint multimod-verify
 	GOOS=darwin GOARCH=amd64 $(GOBUILD) $(LDFLAGS) -o ./build/darwin/aoc_darwin_amd64 ./cmd/awscollector
-	GOOS=linux GOARCH=amd64 $(GOBUILD) $(LDFLAGS) -o ./build/linux/aoc_linux_x86_64 ./cmd/awscollector
```
Comment (sethAmazon) [inline Makefile:81]: Have we talked with @alolita  @anuraaga @Aneurysm9 that being backwards not compatible is okay here

File: cmd/awscollector/Dockerfile [hunk 2]
```diff
@@ -9,16 +9,13 @@ RUN apk --update add ca-certificates
 
 WORKDIR /workspace
 
-ARG TARGETPLATFORM
-RUN echo "Target platform is $TARGETPLATFORM"
```
Comment (sethAmazon) [inline cmd/awscollector/Dockerfile:13]: Think we should echo the target arch we are running for

File: Makefile (hunk 1 above)
Comment (Aneurysm9) [reply Makefile:81]: Is this just an intermediate name used during the build process?  Does the name of the executable included in the `deb`, `rpm`, or container image change?

**Review by Aneurysm9 — COMMENTED** [review]
Is it possible to separate out the lint/formatting changes?  They make it hard to have confidence that the functional changes have all been identified.

**Comment by jefchien** [comment]
> Is it possible to separate out the lint/formatting changes? They make it hard to have confidence that the functional changes have all been identified.

@Aneurysm9  I've separated it out into two commits. The first one should be the actual changes and the second one will be the linting changes.

File: Makefile (hunk 1 above)
Comment (jefchien) [reply Makefile:81]: Yes. This is just an intermediate name. `create_rpm.sh` and `create_deb.sh` both copy the binary in and rename it to `aws-otel-collector`. The Dockerfile also renames it, but to `awscollector`.

File: cmd/awscollector/Dockerfile (hunk 2 above)
Comment (jefchien) [reply cmd/awscollector/Dockerfile:13]: Wouldn't the TARGETARCH just show up in the COPY instruction when it tries to find the file?

**Review by Aneurysm9 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (anuraaga) [reply Makefile:81]: Yeah intermediate names should not need to be preserved.

While we're changing these, can we dedupe, and maybe simplify, by moving into the folder

`./build/linux-amd64/aoc`?

**Review by sethAmazon — APPROVED** [review]

**Pull request merged**